packages/test/test-end-to-end-tests/src/test/payloadSize.spec.ts

```diff
@@ -0,0 +1,85 @@
+ * Licensed under the MIT License.
+ */
+
+import { SharedMap } from "@fluidframework/map";
+import { requestFluidObject } from "@fluidframework/runtime-utils";
+import {
+    ITestFluidObject,
+    ChannelFactoryRegistry,
+    ITestObjectProvider,
+    ITestContainerConfig,
+    DataObjectFactoryType,
+} from "@fluidframework/test-utils";
+import { describeNoCompat } from "@fluidframework/test-version-utils";
+import { IContainer, IErrorBase } from "@fluidframework/container-definitions";
+import { ConfigTypes, IConfigProviderBase } from "@fluidframework/telemetry-utils";
+
+describeNoCompat("Payload size", (getTestObjectProvider) => {
+    const mapId = "mapId";
+    const registry: ChannelFactoryRegistry = [
+        [mapId, SharedMap.getFactory()],
+    ];
+    const testContainerConfig: ITestContainerConfig = {
+        fluidDataObjectType: DataObjectFactoryType.Test,
+        registry,
+    };
+
+    let provider: ITestObjectProvider;
+    beforeEach(() => {
+        provider = getTestObjectProvider();
+    });
+    afterEach(async () => provider.reset());
+
+    let container1: IContainer;
+    let dataObject1: ITestFluidObject;
+    let dataObject1map1: SharedMap;
+
+    const configProvider = ((settings: Record<string, ConfigTypes>): IConfigProviderBase => ({
+        getRawConfig: (name: string): ConfigTypes => settings[name],
+    }));
+
+    const setupContainers = async (
+        containerConfig: ITestContainerConfig,
+        featureGates: Record<string, ConfigTypes> = {},
+    ) => {
+        const configWithFeatureGates = {
+            ...containerConfig,
+            loaderProps: { configProvider: configProvider(featureGates) }
+        };
+
+        container1 = await provider.makeTestContainer(configWithFeatureGates);
+        dataObject1 = await requestFluidObject<ITestFluidObject>(container1, "default");
+        dataObject1map1 = await dataObject1.getSharedObject<SharedMap>(mapId);
+
+        await provider.ensureSynchronized();
+    };
+
+    const generateStringOfSize = (sizeInBytes: number): string => new Array(sizeInBytes + 1).join("0");
+    const setMapKeys = (map: SharedMap, count: number, item: string): void => {
+        for (let i = 0; i < count; i++) {
+            map.set(`key${i}`, item);
+        }
+    };
+
+    const containerError = async (container: IContainer) =>
+        new Promise<IErrorBase | undefined>((resolve) => container.once("closed", (error) => { resolve(error); }));
+
+    // In ODSP/t9s this test should catch the close event with error, as the retries would exhaust the retries.
+    // This cannot be enabled until #8888 is resolved, as locally, there are no payload limits. Note that the timeout
+    // must be increased for this test to more than 10 seconds, which makes enabling this test locally very problematic
+    // in terms of developer experience.
+    it.skip("Cannot send payloads larger than 1MB", async () => {
```

Comment:
I have added this test (or a test along these lines) for a past PR which was closed without merging, it is useful for sending large payloads programmatically to any target (based on the node task running the test). In the first iteration, I believed it was a good idea to add it to the codebase disabled but now I'm having second thoughts (also because there's quite a bit of coverage added by the container runtime unit tests) so I will remove this file in the next iteration unless there's an explicit request to keep it (unlikely).